batchflow/models/torch/efficientnet.py

```diff
@@ -0,0 +1,227 @@
+
+    @classmethod
+    def default_config(cls):
+        """ Define model defaults. """
+        config = super().default_config()
+
+        config['initial_block'] += dict(scalable=True, layout='cna', kernel_size=3, strides=2, filters=32)
+
+        config['body/encoder/num_stages'] = 7
+        config['body/encoder/order'] = ['block']
+        config['body/encoder/blocks'] += dict(base=MBConvBlock,
+                                              n_reps=[1, 2, 2, 3, 3, 4, 1],
+                                              scalable=True,
+                                              kernel_size=[3, 3, 5, 3, 5, 5, 3],
+                                              strides=[1, 2, 2, 2, 1, 2, 1],
+                                              filters=[16, 24, 40, 80, 112, 192, 320],
+                                              expand_ratio=[1, 6, 6, 6, 6, 6, 6],
+                                              se_block=dict(ratio=4))
+
+
+        config['head'] += dict(scalable=True, layout='cna V df', kernel_size=1, strides=1, filters=1280,
+                               dropout_rate=0.2)
+
+        config['common'] += dict(activation=swish, width_factor=1.0, depth_factor=1.0)
+
+        config['loss'] = 'ce'
+
+        return config
+
+    def build_config(self):
+        """ Define model's architecture configuration. """
+        config = super().build_config()
+
+        w_factor = config.get('common/width_factor')
+        d_factor = config.get('common/depth_factor')
+
+        for path in ('initial_block/', 'body/encoder/blocks/', 'head/'):
+            scalable = config.get(path + 'scalable')
+            if scalable:
+                for param, factor in [('filters', w_factor), ('n_reps', d_factor)]:
+                    if factor != 1:
+                        val = config.get(path + param)
+                        if val:
+                            if isinstance(val, int):
+                                val = max(1, int(val * factor))
+                            elif isinstance(val, list):
+                                val = [max(1, int(v * factor)) for v in val]
+                            else:
+                                raise ValueError("{} should be int or list, {} given".format(param, type(val)))
+                            config[path + param] = val
+
+        return config
+
+
+class EfficientNetB1(EfficientNetB0):
+    """ EfficientNetB1 """
+
+    resolution = 240
+
+    @classmethod
+    def default_config(cls):
+        config = super().default_config()
+
+        config['common/width_factor'] = 1.0
+        config['common/depth_factor'] = 1.1
+
+        config['head/dropout_rate'] = 0.2
+
+        return config
+
+
+class EfficientNetB2(EfficientNetB0):
+    """ EfficientNetB2 """
+
+    resolution = 260
+
+    @classmethod
+    def default_config(cls):
+        config = super().default_config()
+
+        config['common/width_factor'] = 1.1
+        config['common/depth_factor'] = 1.2
```

Comment:
Actually, still no change in `n_reps`

Reply:
after changing `int` to `ceil` in rounding statements the issue with number of filters and repetitions being equal for B1 and B2 still exists. 

I use a simplified version of filter_rounding compared to [original code](https://github.com/tensorflow/tpu/blob/4489103eee24cd612ea3c41958c9320c45080c6f/models/official/efficientnet/efficientnet_model.py#L106) but filters and repetitions in B1 and B2 are equal when using the original `round filters` as well

So, these are the details of the original paper: B1 and B2 differ only in resolution and dropout rate

As for using simplified `round_filters` instead of the original function my point was the following. It seems over complicated, and intended for width modifiers < 1, that is reducing filters numbers. Most of extra parameters are not modified anywhere. When calculating filters with all default parameters from the original implementation one gets values that are sometimes bigger and sometimes smaller by same relatively small numbers compared to simple `int(ceil(filters*modifies))` implementation. So I decided to use this simplified version